lib/collection/src/shard/replica_set.rs

```diff
@@ -284,13 +315,15 @@ impl ReplicaSet {
     }
 
     pub(crate) async fn on_optimizer_config_update(&self) -> CollectionResult<()> {
-        if let Some(shard) = &self.local {
-            shard.on_optimizer_config_update().await
+        if let Some(_shard) = &self.local {
+            // shard.on_optimizer_config_update().await
+            todo!("optimize config update on local")
```

Comment:
TODO: Not sure how to implement this one now as local is now a `Box<Shard>`

Reply:
Maybe make a separate enum for local? This would remove `dyn` types with a cost of a little boilerplate

Reply:
Thanks for the idea.

I will try first to implement `on_optimizer_config_update` at the level of the shard enum to see how it goes before :+1: